geode-core/src/test/java/org/apache/geode/cache/query/internal/AttributeDescriptorTest.java

```diff
@@ -219,11 +225,80 @@ public void readReflectionForImplicitMethodInvocationShouldReturnCorrectlyWhenMe
     AttributeDescriptor attributeDescriptor =
         new AttributeDescriptor(typeRegistry, methodInvocationAuthorizer, attributeName);
 
-    Object result = attributeDescriptor.readReflection(testBean);
+    Object result = attributeDescriptor.readReflection(testBean, queryExecutionContext);
     assertThat(result).isInstanceOf(String.class);
     assertThat(result).isEqualTo(attributeName);
   }
 
+  @Test
+  @Parameters({
+      "nonPublicAttributeWithPublicAccessor, nonPublicAttributeWithPublicAccessor",
+      "nonPublicAttributeWithPublicGetterMethod, getNonPublicAttributeWithPublicGetterMethod"})
+  public void readReflectionForImplicitMethodInvocationShouldReturnCorrectlyWhenMethodIsAuthorizedAndCacheResult(
+      String attributeName, String methodName) throws Exception {
+    doReturn(true).when(methodInvocationAuthorizer).authorize(any(), any());
+    AttributeDescriptor attributeDescriptor =
+        new AttributeDescriptor(typeRegistry, methodInvocationAuthorizer, attributeName);
+    attributeDescriptor.readReflection(testBean, queryExecutionContext);
+
+    String cacheKey = TestBean.class.getCanonicalName() + "." + methodName;
+    assertThat(queryExecutionContext.cacheGet(cacheKey)).isNotNull();
```

Comment:
Is this check necessary? Given that the next line asserts that it's equal to true, if it's null the test will still fail without this check, I think?

Reply:
👍